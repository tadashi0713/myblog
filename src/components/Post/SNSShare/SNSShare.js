import React from 'react';
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
  LinkedinShareButton,
  LinkedinIcon,
  LineShareButton,
  LineIcon
} from 'react-share';

const SNSShare = ({ postUrl, postTitle }) => (
  <div>
    <FacebookShareButton url={postUrl}>
      <FacebookIcon size={32} round />
    </FacebookShareButton>
    <TwitterShareButton title={postTitle} url={postUrl}>
      <TwitterIcon size={32} round />
    </TwitterShareButton>
    <LinkedinShareButton url={postUrl}>
      <LinkedinIcon size={32} round />
    </LinkedinShareButton>
    <LineShareButton title={postTitle} url={postUrl}>
      <LineIcon size={32} round />
    </LineShareButton>
  </div>
);

export default SNSShare;
