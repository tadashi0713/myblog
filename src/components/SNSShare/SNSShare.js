import React from 'react';
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
  LinkedinShareButton,
  LinkedinIcon,
  LineShareButton,
  LineIcon
} from 'react-share';
import styles from './SNSShare.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGetPocket } from '@fortawesome/free-brands-svg-icons';

const SNSShare = ({ postUrl, postTitle }) => (
  <ul className={styles.content}>
    <li className={styles.share_button}>
      <FacebookShareButton url={postUrl}>
        <FacebookIcon size={32} round />
      </FacebookShareButton>
    </li>
    <li className={styles.share_button}>
      <TwitterShareButton title={postTitle} url={postUrl}>
        <TwitterIcon size={32} round />
      </TwitterShareButton>
    </li>
    <li className={styles.share_button}>
      <LinkedinShareButton url={postUrl}>
        <LinkedinIcon size={32} round />
      </LinkedinShareButton>
    </li>
    <li className={styles.share_button}>
      <LineShareButton title={postTitle} url={postUrl}>
        <LineIcon size={32} round />
      </LineShareButton>
    </li>
    <li className={styles.share_button}>
      <a
        aria-label="share_pocket"
        className={styles.pocket_icon_link}
        href={`https://getpocket.com/edit?url=${postUrl}&title=${postTitle}`}
        onClick={`window.open(this.href, 'PCwindow', 'width=550, height=350, menubar=no, toolbar=no, scrollbars=yes'); return false;`}>
        <FontAwesomeIcon
          color="#fff"
          className={styles.pocket_icon}
          icon={faGetPocket} />
      </a>
    </li>
  </ul>
);

export default SNSShare;
