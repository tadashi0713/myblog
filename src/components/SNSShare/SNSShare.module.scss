.snsshare {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: 0 0.8rem 0 0;
    line-height: 1;
    height: 20px;
  }
}
