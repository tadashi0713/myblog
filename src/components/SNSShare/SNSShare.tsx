import React from "react";
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
  LinkedinShareButton,
  LinkedinIcon,
  LineShareButton,
  LineIcon
} from "react-share";
import Helmet from "react-helmet";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faGetPocket } from "@fortawesome/free-brands-svg-icons";
import styles from "./SNSShare.module.scss";

const SNSShare = ({ postUrl, postTitle }) => (
  <ul className={styles.content}>
    <Helmet>
      <script
        type="text/javascript"
        src="//widgets.getpocket.com/v1/j/btn.js?v=1"
        charSet="utf-8"
        async={true}
      />
    </Helmet>
    <li className={styles.share_button}>
      <FacebookShareButton
        additionalProps={{ "aria-label": "share_facebook" }}
        url={postUrl}
      >
        <FacebookIcon size={32} round />
      </FacebookShareButton>
    </li>
    <li className={styles.share_button}>
      <TwitterShareButton
        additionalProps={{ "aria-label": "share_twitter" }}
        title={postTitle}
        url={postUrl}
      >
        <TwitterIcon size={32} round />
      </TwitterShareButton>
    </li>
    <li className={styles.share_button}>
      <LinkedinShareButton
        additionalProps={{ "aria-label": "share_linkedin" }}
        url={postUrl}
      >
        <LinkedinIcon size={32} round />
      </LinkedinShareButton>
    </li>
    <li className={styles.share_button}>
      <LineShareButton
        additionalProps={{ "aria-label": "share_line" }}
        title={postTitle}
        url={postUrl}
      >
        <LineIcon size={32} round />
      </LineShareButton>
    </li>
    <li className={styles.share_button}>
      <a
        aria-label="share_pocket"
        className={styles.pocket_icon_link}
        href={`https://getpocket.com/edit?url=${postUrl}&title=${postTitle}`}
        onClick={() =>
          window.open(
            this.href,
            "PCwindow",
            "width=550, height=350, menubar=no, toolbar=no, scrollbars=yes"
          )
        }
      >
        <FontAwesomeIcon
          color="#fff"
          className={styles.pocket_icon}
          icon={faGetPocket}
        />
      </a>
    </li>
  </ul>
);

export default SNSShare;
