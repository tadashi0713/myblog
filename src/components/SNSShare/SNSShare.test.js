import React from 'react';
import renderer from 'react-test-renderer';
import SNSShare from './SNSShare';

describe('SNSShare', () => {
  const props = {
    postTitle: 'Sample',
    postUrl: 'https://tadashi0713.com/'
  };

  it('renders correctly', () => {
    const tree = renderer.create(<SNSShare {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
