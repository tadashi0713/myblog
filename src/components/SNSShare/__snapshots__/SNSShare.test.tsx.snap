// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNSShare renders correctly 1`] = `
<div
  className="snsshare"
>
  <li>
    <div
      className="fb-share-button"
      data-href="https://tadashi0713.com/sample"
      data-layout="button_count"
      data-size="small"
    />
  </li>
  <li>
    <a
      className="twitter-share-button"
      data-show-count="false"
      data-text="Sample"
      data-url="https://tadashi0713.com/sample"
      href="https://twitter.com/share?ref_src=twsrc%5Etfw"
    >
      Tweet
    </a>
  </li>
  <li>
    <script
      src="https://platform.linkedin.com/in.js"
      type="text/javascript"
    >
      lang: ja_JP
    </script>
    <script
      data-url="https://tadashi0713.com/sample"
      type="IN/Share"
    />
  </li>
  <li>
    <a
      className="hatena-bookmark-button"
      data-hatena-bookmark-height="20"
      data-hatena-bookmark-lang="ja"
      data-hatena-bookmark-layout="basic-counter"
      href="https://b.hatena.ne.jp/entry/s/tadashi0713.com/sample"
      title="このエントリーをはてなブックマークに追加"
    >
      <img
        alt="このエントリーをはてなブックマークに追加"
        height="20"
        src="https://b.st-hatena.com/images/v4/public/entry-button/button-only@2x.png"
        style={
          Object {
            "border": "none",
          }
        }
        width="20"
      />
    </a>
  </li>
</div>
`;
